app/Keycloak/Listeners/BlockClients.php

```diff
@@ -19,24 +19,22 @@ final class BlockClients implements ShouldQueue
     use Queueable;
 
     public function __construct(
-        private readonly IntegrationRepository $integrationRepository,
         private readonly KeycloakClientRepository $keycloakClientRepository,
         private readonly ApiClient $client,
         private readonly LoggerInterface $logger
     ) {
     }
 
-    public function handle(IntegrationBlocked $integrationBlocked): void
+    public function handle(IntegrationBlocked|IntegrationDeleted $integrationBlocked): void
     {
-        $integration = $this->integrationRepository->getById($integrationBlocked->id);
```

Comment:
FYI: it's fine like it is now, but there is a `getByIdWithTrashed` to get deleted integration.